File: src/MessageFlow/UnhandledRequestException.cs
namespace MessageFlow;

/// <summary>
/// Thrown when no handler of a chain accepted the request and no fallback was configured.
/// </summary>
public sealed class UnhandledRequestException : InvalidOperationException
{
    /// <summary>
    /// Initializes a new instance of the <see cref="UnhandledRequestException"/> class.
    /// </summary>
    public UnhandledRequestException()
        : base("No handler in the chain handled the request.")
    {
    }

    /// <summary>
    /// Initializes a new instance of the <see cref="UnhandledRequestException"/> class.
    /// </summary>
    /// <param name="message">The message describing the error.</param>
    public UnhandledRequestException(string message)
        : base(message)
    {
    }

    /// <summary>
    /// Initializes a new instance of the <see cref="UnhandledRequestException"/> class.
    /// </summary>
    /// <param name="message">The message describing the error.</param>
    /// <param name="innerException">The exception that caused this error.</param>
    public UnhandledRequestException(string message, Exception innerException)
        : base(message, innerException)
    {
    }
}

File: java/src/main/java/io/github/messageflow/UnhandledRequestException.java
package io.github.messageflow;

/**
 * Thrown when no handler of a chain accepted the request and no fallback was configured.
 */
public final class UnhandledRequestException extends IllegalStateException {

    private static final long serialVersionUID = 1L;

    /** Creates an exception with the default message. */
    public UnhandledRequestException() {
        super("No handler in the chain handled the request.");
    }

    /**
     * Creates an exception with the given message.
     *
     * @param message the message describing the error
     */
    public UnhandledRequestException(String message) {
        super(message);
    }

    /**
     * Creates an exception with the given message and cause.
     *
     * @param message the message describing the error
     * @param cause   the exception that caused this error
     */
    public UnhandledRequestException(String message, Throwable cause) {
        super(message, cause);
    }
}
